src/redux/actions/projectActions.ts

```diff
@@ -24,11 +28,35 @@ export function loadProject(project: IProject): (dispatch: Dispatch) => Promise<
     };
 }
 
-export function saveProject(project: IProject): (dispatch: Dispatch) => Promise<IProject> {
-    return async (dispatch: Dispatch) => {
+export function saveProject(project: IProject):
+  (dispatch: Dispatch, getState: any) => Promise<IProject> {
+    return async (dispatch: Dispatch, getState: any) => {
         const projectService = new ProjectService();
-        project = await projectService.save(project);
-        dispatch(saveProjectAction(project));
+        const projectId = project["id"];
+        const projectName = project["name"];
+        const sourceConnection = project.sourceConnection.name;
+        const targetConnection = project.targetConnection.name;
+        const projectList = getState().recentProjects;
+        if (projectList && projectList.length > 0) {
```

Comment:
Let's wrap up this duplicate check and move it into its own function to keep the core saveProject method clean and concise.  I think it actually would fit best inside the projectService

Reply:
Actually as I look at this, that would make the logic much easier to implement. Thanks, I'll be working on that today :)